### PR TITLE
Support c_void_ptr -> c_ptr assignments and stop treating c_ptr as a class

### PR DESCRIPTION
**Pull request by bradcray**
This PR makes two simple changes:

1) It adds support for assigning from c_void_ptr to c_ptr
2) It has isClassType() return `false` for `c_ptr` types (without this second change, the first resulted in an ambiguity error

It also adds two simple tests to lock in this behavior.

This test fixes the issue reported in issue #10177 though I'm not sure how it's related to the [observed behavior](https://github.com/chapel-lang/chapel/issues/10177#issuecomment-404518430) there in which a SymExpr is missing its parentExpr causing an assertion failure.  I'm going to hold off on closing that issue until we understand the failure mode better to make sure we're not papering over a bigger problem; and to verify that this helps fix things for the larger code which motivated its simplified test case.

### CODE REVIEW

File: compiler/resolution/preFold.cpp [hunk 1]
```diff
@@ -456,7 +456,8 @@ static Expr* preFoldPrimOp(CallExpr* call) {
   } else if (call->isPrimitive(PRIM_IS_CLASS_TYPE)) {
     Type* t = call->get(1)->typeInfo();
 
-    if (isClassLike(t) && !t->symbol->hasFlag(FLAG_EXTERN)) {
+    if (isClassLike(t) && !t->symbol->hasFlag(FLAG_EXTERN) &&
+        !t->symbol->hasFlag(FLAG_C_PTR_CLASS)) {
```
Comment (mppf) [inline compiler/resolution/preFold.cpp]: Should it also rule out _ddata (FLAG_DATA_CLASS)?

File: modules/internal/CPtr.chpl [hunk 2]
```diff
@@ -106,6 +106,9 @@ module CPtr {
   pragma "no doc"
   inline proc =(ref a:c_ptr, b:c_ptr) { __primitive("=", a, b); }
 
+  pragma "no doc"
+  inline proc =(ref a: c_ptr, b: c_void_ptr) { __primitive("=", a, b); }
+
```
Comment (vasslitvinov) [inline modules/internal/CPtr.chpl]: How about going in the other direction, `=(c_void_ptr, c_ptr)` ? Seems it is equally - or even more - valid.

**Comment by vasslitvinov** [comment]
With the disclaimer that I don't use these features, this looks good to me. 
I suggest allowing `=(c_void_ptr, c_ptr)` as well.

File: modules/internal/CPtr.chpl (hunk 2 above)
Comment (mppf) [reply modules/internal/CPtr.chpl]: That should already work because we allow coercions from c_ptr to c_void_ptr.

Comment (bradcray) [reply modules/internal/CPtr.chpl]: Trying a quick test, it seems it does.

**Comment by vasslitvinov** [comment]
@bradcray - it is OK to go ahead and merge this PR before #10177 is fixed.
10177 still crashes - just need `--no-local`.

**Comment by bradcray** [comment]
> #10177 still crashes - just need `--no-local`.

D'oh!

**Comment by bradcray** [comment]
(that said, I was just waiting to see if anybody squawked at the design, less because of the bug... but thanks for pointing out my error).

File: compiler/resolution/preFold.cpp (hunk 1 above)
Comment (bradcray) [reply compiler/resolution/preFold.cpp]: I'm gonna leave that to a separate PR just to keep this one simple / focused and leverage the testing I've already done.  But I'll check it out.

**Pull request merged**